lxd/device/disk.go

```diff
@@ -439,6 +461,44 @@ func (d *disk) startVM() (*deviceConfig.RunConfig, error) {
 						return nil, err
 					}
 				}
+
+				poolID, err := d.state.Cluster.GetStoragePoolID(d.config["pool"])
+				if err != nil {
+					return nil, err
+				}
+
+				// Only custom volumes can be attached currently.
+				storageProjectName, err := project.StorageVolumeProject(d.state.Cluster, d.inst.Project(), db.StoragePoolVolumeTypeCustom)
+				if err != nil {
+					return nil, err
+				}
+
+				_, volume, err := d.state.Cluster.GetLocalStoragePoolVolume(storageProjectName, d.config["source"], db.StoragePoolVolumeTypeCustom, poolID)
+				if err != nil {
+					return nil, err
+				}
+
+				contentType, err := storagePools.VolumeContentTypeNameToContentType(volume.ContentType)
+				if err != nil {
+					return nil, err
+				}
+
+				if contentType == db.StoragePoolVolumeContentTypeBlock {
+					pool, err := storagePools.GetPoolByName(d.state, d.config["pool"])
+					if err != nil {
+						return nil, err
+					}
+
+					// Get the volume name on storage.
+					volStorageName := project.StorageVolume(storageProjectName, volume.Name)
+
+					vol := storageDrivers.NewVolume(pool.Driver(), pool.Name(), storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeBlock, volStorageName, nil, nil)
+
+					srcPath, err = pool.Driver().GetVolumeDiskPath(vol)
+					if err != nil {
+						return nil, err
+					}
```

Comment:
I'm confused, why all this duplicated logic rather than pushing this into mountPoolVolume?

Reply:
fixed